### PR TITLE
Fix two-column scan statistics

### LINKED ISSUE #1132: Two column non-equality scan statistics may be incorrect
#866 introduced a problem for scans on two columns with a different predicate than `Equals`.
The early exit introduced there only makes sense for equality predicates, since a non-overlapping range can still yield results for other predicates.
I haven't checked whether it is as simple as changing it to 

```c++
if (overlapping_range_min > overlapping_range_max 
    && predicate_condition == PredicateCondition::Equals) {
  ...
}
```

or if that possibly introduces the bug in #865 or others again.

### PR DESCRIPTION
**Pull request by timzimmermann**
Fixes #1132.

### CODE REVIEW

File: src/lib/statistics/column_statistics.cpp [hunk 1]
```diff
@@ -247,20 +242,12 @@ FilterByColumnComparisonEstimate ColumnStatistics<ColumnDataType>::estimate_pred
   auto right_above_overlapping_ratio = 0.f;
 
   if (std::is_integral<ColumnDataType>::value) {
```
Comment (FabianWiebe) [inline src/lib/statistics/column_statistics.cpp]: Can be also replaced to `if constexpr (std::is_integral_v<ColumnDataType>) {`.
Makes it consistent with the other if condition.

Comment (timzimmermann) [reply src/lib/statistics/column_statistics.cpp]: Good catch, thanks, I changed it in two more places as well.

File: src/lib/statistics/column_statistics.cpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ FilterByValueEstimate ColumnStatistics<ColumnDataType>::estimate_predicate_with_
       // distinction between integers and floats
       // for integers "< value" means that the new max is value <= value - 1
       // for floats "< value" means that the new max is value <= value - ε
-      if (std::is_integral_v<ColumnDataType>) {
+      if constexpr (std::is_integral_v<ColumnDataType>) {
```
Comment (mrks) [inline src/lib/statistics/column_statistics.cpp]: You'll probably get a false positive error with clang-tidy here, at least until we upgrade to LLVM 7. Use NOLINT if necessary

Comment (timzimmermann) [reply src/lib/statistics/column_statistics.cpp]: Neat, I did not know clang-tidy supports NOLINT as well.

**Pull request merged**